assets/sass/layout/_header.scss

```diff
@@ -92,20 +95,54 @@
 					}
 				}
 			}
+
+			
 		}
 	}
 
 	.navbar-dropdown {
 		div.dropdown-backdrop {
 			display: none;
 		}
-		@media (max-width: $grid-float-breakpoint-max) {
-			li.open > ul.dropdown-menu {
-				& > li > a{
-					color: #dadada;
+
+		@media (max-width: $screen-md-max) {
+			li.dropdown > a.dropdown-toggle {
+				max-width: $screen-md-max - 700px;
+				white-space: nowrap;
+				overflow: hidden;
+				text-overflow: ellipsis;
+			}
+		}
+
+		@media (max-width: $screen-sm-max) {
+			li.dropdown > a.dropdown-toggle {
+				max-width: $screen-sm-max - 720px;
+			}
+		}
+
+		@media (max-width: $screen-xs-max) {
+			li.dropdown > a.dropdown-toggle {
+				max-width: $screen-xs-max - 600px;
+			}
```

Comment:
I think the subtracted pixel values could be somewhat smaller. The values 700px, 720px, and 600px currently leave a lot of empty space to the right side of the course name.

Also, I noticed that the old, unwanted behavior can still occur with large screens (`$screen-lg-max`).